### PR TITLE
Fix copysign bug

### PR DESCRIPTION
**Pull request by zxh0**
According to the [Wasm spec](https://webassembly.github.io/spec/core/exec/numerics.html#op-fcopysign), it seems that f32Copysign and f64Copysign was not correctly implemented. I did some fix and unit tests, but please double check this.

### CODE REVIEW

File: exec/num_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package exec
```
Comment (sbinet) [inline exec/num_test.go]: please add the necessay copyright blurbl.

File: exec/num_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"testing"
+
+	"github.com/go-interpreter/wagon/wasm/operators"
+)
+
+func TestF32BinOps(t *testing.T) {
+	testF32BinOp(t, operators.F32Sub, 3.0, 2.0, 1.0)
```
Comment (sbinet) [inline exec/num_test.go]: please modify to use a "table-driven" testcases approach.

File: exec/num_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	testF32BinOp(t, operators.F32Copysign, 3.0, 2.0, 3.0)
+	testF32BinOp(t, operators.F32Copysign, 3.0, -2.0, -3.0)
+}
+
+func TestF64BinOps(t *testing.T) {
+	testF64BinOp(t, operators.F64Sub, 3.0, 2.0, 1.0)
```
Comment (sbinet) [inline exec/num_test.go]: please modify to use a "table-driven" testcases approach.

File: exec/num_test.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+func testF32BinOp(t *testing.T, opcode byte, z1, z2, z3 float32) {
+	vm := &VM{}
+	vm.newFuncTable()
+	vm.pushFloat32(z1)
+	vm.pushFloat32(z2)
+	vm.funcTable[opcode]()
+	if r := vm.popFloat32(); r != z3 {
+		op, _ := operators.New(opcode)
```
Comment (sbinet) [inline exec/num_test.go]: please add a check for the returned error.

File: exec/num_test.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+func testF64BinOp(t *testing.T, opcode byte, z1, z2, z3 float64) {
+	vm := &VM{}
+	vm.newFuncTable()
+	vm.pushFloat64(z1)
+	vm.pushFloat64(z2)
+	vm.funcTable[opcode]()
+	if r := vm.popFloat64(); r != z3 {
+		op, _ := operators.New(opcode)
```
Comment (sbinet) [inline exec/num_test.go]: please add a check for the returned error.

**Comment by zxh0** [comment]
Thank you for the nice comments. I have add some explanatory and changed the code.

File: exec/num_test.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The go-interpreter Authors.  All rights reserved.
```
Comment (sbinet) [inline exec/num_test.go]:
```suggestion
// Copyright 2020 The go-interpreter Authors.  All rights reserved.
```

File: exec/num_test.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+package exec
+
+import (
+	"testing"
+
+	"github.com/go-interpreter/wagon/wasm/operators"
+)
+
+func TestF32BinOps(t *testing.T) {
```
Comment (sbinet) [inline exec/num_test.go]: this should probably be rewritten as:

```go
func TestF32BinOps(t *testing.T) {
	for _, tc := range []struct {
		opcode byte
		z1     float32
		z2     float32
		want   float32
	}{
		{operators.F32Sub, 3.0, 2.0, 1.0},
		{operators.F32Copysign, 3.0, 2.0, 3.0},
		{operators.F32Copysign, 3.0, -2.0, -3.0},
	} {
		name, err := operators.New(tc.opcode)
		if err != nil {
			t.Fatalf("could not lookup operator 0x%x: %+v", tc.opcode, name)
		}
		t.Run(fmt.Sprintf("%v(%v,%v)", name, tc.z1, tc.z2), func(t *testing.T) {
			vm := new(VM)
			vm.newFuncTable()
			vm.pushFloat32(tc.z1)
			vm.pushFloat32(tc.z2)
			vm.funcTable[test.opcode]()
			if got, want := vm.popFloat32(), tc.want; got != want {
				t.Fatalf("got=%v, want=%v", got, want)
			}
		})
	}
}
```

File: exec/num_test.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+				t.Error(err.Error())
+			} else {
+				t.Errorf("%s(%f, %f) = %f (!= %f)",
+					op.Name, test.z1, test.z2, result, test.result)
+			}
+		}
+	}
+}
+
+func TestF64BinOps(t *testing.T) {
```
Comment (sbinet) [inline exec/num_test.go]: same applies here.

**Comment by zxh0** [comment]
Done for the `AUTHORS` & `CONTRIBUTORS`. Thank you again for this Wasm implementation.

**Pull request merged**